bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/GC.java

```diff
@@ -968,6 +968,7 @@ public void drawImage (Image image, int srcX, int srcY, int srcWidth, int srcHei
 	}
 	if (image == null) SWT.error (SWT.ERROR_NULL_ARGUMENT);
 	if (image.isDisposed()) SWT.error(SWT.ERROR_INVALID_ARGUMENT);
+	image.handleDPIChange(DPIUtil.getDeviceZoom());
```

Comment:
Why is this call necessary? At the end, this method calls another method that also does the rescaling. I am concerned that this change will have an unknown impact on the "hack" applied later on in this method.

Reply:
actually, it is only necessary because of that hack later. But having a second look on it, I was able to simplify the whole call by replacing `getBoundsInPixels()` with `getBounds(deviceZoom) `